.github/workflows/main.yml

```diff
@@ -36,8 +38,41 @@ jobs:
     strategy:
       matrix:
         include:
-          - python: 3.9
-            wagtail: wagtail>=4.1,<5.1
+          - python: "3.9"
+            django: django>=3.2,<4.0
+            wagtail: wagtail>=4.1,<4.2
+          - python: "3.9"
+            django: django>=4.1,<4.2
+            wagtail: wagtail>=4.2,<5.0
+          - python: "3.9"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.0,<5.1
+          - python: "3.9"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.1,<5.2
+
+          - python: "3.10"
+            django: django>=3.2,<4.0
+            wagtail: wagtail>=4.1,<4.2
+          - python: "3.10"
+            django: django>=4.1,<4.2
+            wagtail: wagtail>=4.2,<5.0
+          - python: "3.10"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.0,<5.1
+          - python: "3.10"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.1,<5.2
+
+          - python: "3.11"
+            django: django>=4.1,<4.2
+            wagtail: wagtail>=4.2,<5.0
+          - python: "3.11"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.0,<5.1
+          - python: "3.11"
+            django: django>=4.2,<4.3
+            wagtail: wagtail>=5.1,<5.2
```

Comment:
Should we make this complete? If not necessary, the above is OK

```suggestion
          - python: "3.9"
            django: django>=3.2,<4.0
            wagtail: wagtail>=4.1,<4.2
          - python: "3.9"
            django: django>=3.2,<4.0
            wagtail: wagtail>=4.2,<5.0
          - python: "3.9"
            django: django>=3.2,<4.0
            wagtail: wagtail>=5.0,<5.1
          - python: "3.9"
            django: django>=3.2,<4.0
            wagtail: wagtail>=5.1,<5.2

           - python: "3.9"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.1,<4.2
          - python: "3.9"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.2,<5.0
          - python: "3.9"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.0,<5.1
          - python: "3.9"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.1,<5.2

           - python: "3.10"
            django: django>=3.2,<4.0
            wagtail: wagtail>=4.1,<4.2
          - python: "3.10"
            django: django>=3.2,<4.0
            wagtail: wagtail>=4.2,<5.0
          - python: "3.10"
            django: django>=3.2,<4.0
            wagtail: wagtail>=5.0,<5.1
          - python: "3.10"
            django: django>=3.2,<4.0
            wagtail: wagtail>=5.1,<5.2

           - python: "3.10"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.1,<4.2
          - python: "3.10"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.2,<5.0
          - python: "3.10"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.0,<5.1
          - python: "3.10"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.1,<5.2

           - python: "3.11"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.1,<4.2
          - python: "3.11"
            django: django>=4.1,<4.2
            wagtail: wagtail>=4.2,<5.0
          - python: "3.11"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.0,<5.1
          - python: "3.11"
            django: django>=4.1,<4.2
            wagtail: wagtail>=5.1,<5.2
            
          - python: "3.11"
            django: django>=4.2,<4.3
            wagtail: wagtail>=5.0,<5.1
          - python: "3.11"
            django: django>=4.2,<4.3
            wagtail: wagtail>=5.1,<5.2
```

Reply:
I didn't want to add every possible permutation here because the original setup didn't do that and I thought there could be a reason for that. It might have been to have a reasonably quick testing time. For now I'll leave it alone but thanks @katdom13